File: src/main/resources/output/IFC4X3_python/Entities/IfcMaterialDefinition/IfcMaterialLayer/IfcMaterialLayer.py
class IfcMaterialLayer(IfcMaterialDefinition):
    def __init__(self):
        self.Material: IfcMaterial = None
        self.LayerThickness: IfcNonNegativeLengthMeasure = None
        self.IsVentilated: IfcLogical = None
        self.Name: IfcLabel = None
        self.Description: IfcText = None
        self.Category: IfcLabel = None
        self.Priority: IfcInteger = None

    # === EXTENDED BY ===
    # IfcMaterialLayerWithOffsets

    # === INVERSE CLAUSES ===
    # ToMaterialLayerSet : IfcMaterialLayerSet FOR MaterialLayers

    # === WHERE CLAUSES ===
    # NormalizedPriority : NOT(EXISTS(Priority)) OR {0 <= Priority <= 100}

File: src/main/resources/output/IFC4X3_csharp/Entities/IfcMaterialDefinition/IfcMaterialLayer/IfcMaterialLayer.cs
public class IfcMaterialLayer : IfcMaterialDefinition
{
    public IfcMaterial Material { get; set; }
    public IfcNonNegativeLengthMeasure LayerThickness { get; set; }
    public IfcLogical IsVentilated { get; set; }
    public IfcLabel Name { get; set; }
    public IfcText Description { get; set; }
    public IfcLabel Category { get; set; }
    public IfcInteger Priority { get; set; }

    // === EXTENDED BY ===
    // IfcMaterialLayerWithOffsets

    // === INVERSE CLAUSES ===
    // ToMaterialLayerSet : IfcMaterialLayerSet FOR MaterialLayers

    // === WHERE CLAUSES ===
    // NormalizedPriority : NOT(EXISTS(Priority)) OR {0 <= Priority <= 100}
}
